java/vector/src/test/java/org/apache/arrow/vector/TestVectorReAlloc.java

```diff
@@ -191,6 +189,14 @@ public void testVariableAllocateAfterReAlloc() throws Exception {
     }
   }
 
+  @Test
+  public void test() throws Exception {
+    try (final VarCharVector vector = new VarCharVector("", allocator)) {
+      vector.allocateNew(6000);
+      Assert.assertEquals(1000, vector.getValidityBuffer().capacity() );
```

Comment:
can we make the 

Assert.assertTrue(vector.getValidityBuffer().capacity() >= DataSizeRoundingUtil.divideBy8Ceil(6000));

The APIs only guarantee that the capacity is at least for 6000 elements.